### ISSUE TITLE
Sending blank message visually appears in the Web UI

### ISSUE THREAD

**Issue by xlash123** [iandwelker/smserver#39]
Either by pressing enter or clicking the send button while the input field is blank, an empty message bubble appears. This is only a visual bug, and no blank message was actually sent.

I am running SMServer version 0.4.2 with libsmserver version 0.3.5 on iOS 13.7 on iPhone 7 using odysseyra1n.

**Comment by iandwelker** [comment]
Thanks for letting me know about this; I've fixed the issue in my code. Version 0.4.3, when I release it, will have the fix included.

**Comment by iandwelker** [comment]
0.4.3 has been released. It should fix this issue, but let me know if it doesn't.

**Comment by londondev77** [comment]
Just did a quick test and found a small bug. Not sure if it was a one-off or will repeat.

Here is an image of the web interface:
![2020-09-20 19_53_33-Window](https://user-images.githubusercontent.com/1478538/93719621-ab32be00-fb7b-11ea-87ef-37fb119a9843.png)

And then from iMessage:
![20200920_185450000_iOS](https://user-images.githubusercontent.com/1478538/93719646-df0de380-fb7b-11ea-9381-1cd0ceeb4045.jpg)

Note that on the web, one message is repeated twice instead of showing my reply.

Here's a snippet from the console log:
https://pastebin.com/44VfEQTu

I'll keep using it to see if it happens again.

**Comment by xlash123** [comment]
The blank text is fixed, however now I just see it duplicating the message last (both sent by me and sent to me). The `send` API is returning false, but `requests` is returning the message that was last sent. Both are still just visual bugs in the UI.

**Comment by iandwelker** [comment]
So just to clarify: You are able to send texts via the web interface and they successfully get sent to the person on the other end, but when you send a post request to the `/send` subdirectory, it returns false despite successfully sending the text?

Also, if you look at some of the recent comments in issue #37, you'll see that some others are having this same issue as well. However, as you'll also see, I can't work on this right now due to dependency issues, so the fix for this may have to wait a bit. I'll update once I definitely figure it out.

**Comment by xlash123** [comment]
What I mean is sending a blank text triggers the duplication, and `/send` returns false and `/requests` returns the last message sent/received. This seems unrelated to #37 as this happens to me reliably on sending a blank message.

**Comment by iandwelker** [comment]
Ohhh, ok I'm following; thank you for pointing that out. And yes, that makes a lot of sense; I know where the issue is now, and I'll have it fixed by next version (whenever that comes out). If you would like to fix it yourself so that you don't keep running into the same issue, feel free to open `/Applications/SMServer.app/html/chats.html` and edit it like so:

On line 605, add the following:
```javascript
var get_new = (oldval.length > 0 && selected.length > 0 && document.getElementById('attachmentbutton').files.length > 0);
```

Then, change lines 619-623 to say the following: 
```javascript
if (get_new) {
    await sleep(500);

    getLatestText();
    setChatAsTop(current_chat_id, oldval);
}
```

**Comment by iandwelker** [comment]
This should be completely fixed with version 0.5.0. Can you verify?

**Comment by iandwelker** [comment]
This task has been note received any updates from the original creator for quite a while, and has, as far as I can tell, been fixed. I'm closing it now; feel free to reopen if it has not actually been fixed.
